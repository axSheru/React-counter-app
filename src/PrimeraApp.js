import React from 'react';
import PropTypes from 'prop-types';
//Functional Component.
const PrimeraApp = ({ saludo }) => {

    return(
        <>
            <h1>{ saludo }</h1>
            <p>Mi primera aplicaci√≥n.</p>
        </>
    );
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
};

export default PrimeraApp;