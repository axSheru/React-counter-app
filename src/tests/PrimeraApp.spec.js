import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en PrimeraApp.', () => {

    /* test('debe de retornar saludo.', () => {

        const saludo = 'Hola, soy Luke!';

        const { getByText } = render(<PrimeraApp saludo={ saludo }/>);

        expect( getByText( saludo ) ).toBeInTheDocument();

    }); */

    test('debe renderizar PrimeraApp correctamente.', () => {

        const saludo = 'Hola, soy Luke!';

        const wrapper = shallow(<PrimeraApp saludo={ saludo } />);

        expect( wrapper ).toMatchSnapshot();

    });

});