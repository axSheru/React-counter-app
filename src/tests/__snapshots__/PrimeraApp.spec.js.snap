// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp. debe renderizar PrimeraApp correctamente. 1`] = `
<Fragment>
  <h1>
    Hola, soy Luke!
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
